### PR TITLE
URI Escape next-marker and prefix query params for get_object

### PR DESCRIPTION
**Pull request by DenverM80**

### CODE REVIEW

File: ds3-sdk-integration/src/test/java/com/spectralogic/ds3client/integration/Regression_Test.java [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+            final List<Ds3Object> objects = Lists.newArrayList(
+                    new Ds3Object("obj1_no_spaces.txt", 1024),
+                    new Ds3Object("obj2_no_spaces.txt", 1024),
+                    new Ds3Object("obj3 has spaces.txt", 1024));
+
+            final Ds3ClientHelpers.Job putJob = helpers.startWriteJob(bucketName, objects);
+
+            putJob.transfer(new Ds3ClientHelpers.ObjectChannelBuilder() {
+                @Override
+                public SeekableByteChannel buildChannel(String key) throws IOException {
```
Comment (rpmoore) [inline ds3-sdk-integration/src/test/java/com/spectralogic/ds3client/integration/Regression_Test.java]: Missing final

File: ds3-sdk/src/main/java/com/spectralogic/ds3client/commands/GetBucketRequest.java [hunk 2]
```diff
@@ -52,7 +53,7 @@ public GetBucketRequest withNextMarker(final String nextMarker) {
      */
     public GetBucketRequest withPrefix(final String prefix) {
         this.prefix = prefix;
-        this.getQueryParams().put("prefix", prefix);
+        this.getQueryParams().put("prefix", UrlEscapers.urlFragmentEscaper().escape(prefix));
```
Comment (rpmoore) [inline ds3-sdk/src/main/java/com/spectralogic/ds3client/commands/GetBucketRequest.java]: I don't see a unit test for this line.  We probably need one to make sure the escaping behavior that we assume needs to be here, is actually needed.

**Comment by DenverM80** [comment]
Updated diff after review comments.

File: ds3-sdk/src/main/java/com/spectralogic/ds3client/helpers/Ds3ClientHelpersImpl.java [hunk 3]
```diff
@@ -155,16 +155,21 @@ public void ensureBucketExists(final String bucket) throws IOException, Signatur
 
     @Override
     public Iterable<Contents> listObjects(final String bucket, final String keyPrefix) throws SignatureException, IOException {
-        return this.listObjects(bucket, keyPrefix, Integer.MAX_VALUE);
+        return this.listObjects(bucket, keyPrefix, null, Integer.MAX_VALUE);
+    }
```
Comment (rpmoore) [inline ds3-sdk/src/main/java/com/spectralogic/ds3client/helpers/Ds3ClientHelpersImpl.java]: Missing newline between methods

File: ds3-sdk-integration/src/test/java/com/spectralogic/ds3client/integration/Regression_Test.java [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
+                    final ByteArraySeekableByteChannel channel = new ByteArraySeekableByteChannel(1024);
+                    channel.write(randomBuffer);
+
+                    return channel;
+                }
+            });
+
+            final Iterable<Contents> objs = helpers.listObjects(bucketName, null, "obj3 has spaces.txt");
+            boolean foundObj4 = false;
+            for( Contents obj : objs ) {
```
Comment (rpmoore) [inline ds3-sdk-integration/src/test/java/com/spectralogic/ds3client/integration/Regression_Test.java]: Missing final

File: ds3-sdk-integration/src/test/java/com/spectralogic/ds3client/integration/Regression_Test.java [hunk 5]
```diff
@@ -0,0 +1,133 @@
...
+                    new Ds3Object("obj1_no_spaces.txt", 1024),
+                    new Ds3Object("has spaces obj2.txt", 1024),
+                    new Ds3Object("obj3_no_spaces.txt", 1024),
+                    new Ds3Object("has spaces obj4.txt", 1024));
+
+            final Ds3ClientHelpers.Job putJob = helpers.startWriteJob(bucketName, objects);
+
+            putJob.transfer(new Ds3ClientHelpers.ObjectChannelBuilder() {
+                @Override
+                public SeekableByteChannel buildChannel(String key) throws IOException {
```
Comment (rpmoore) [inline ds3-sdk-integration/src/test/java/com/spectralogic/ds3client/integration/Regression_Test.java]: Missing final

File: ds3-sdk-integration/src/test/java/com/spectralogic/ds3client/integration/Regression_Test.java [hunk 6]
```diff
@@ -0,0 +1,133 @@
...
+                    channel.write(randomBuffer);
+
+                    return channel;
+                }
+            });
+
+            final Iterable<Contents> objs = helpers.listObjects(bucketName, "has spaces");
+            boolean foundObj2 = false;
+            boolean foundObj4 = false;
+            for( Contents obj : objs ) {
```
Comment (rpmoore) [inline ds3-sdk-integration/src/test/java/com/spectralogic/ds3client/integration/Regression_Test.java]: Missing final

File: ds3-sdk/src/main/java/com/spectralogic/ds3client/commands/GetBucketRequest.java (hunk 2 above)
Comment (DenverM80) [inline ds3-sdk/src/main/java/com/spectralogic/ds3client/commands/GetBucketRequest.java]: testPrefixWithSpaces added.

**Pull request merged**